sdk/js/src/Client.ts

```diff
@@ -300,22 +301,32 @@ export class DVCClient implements Client {
     }
 
     private registerVisibilityChangeHandler() {
-        if (typeof document !== 'undefined') {
-            this.pageVisibilityHandler = () => {
-                if (document.visibilityState === 'visible') {
+        if (typeof document === 'undefined') {
+            return
+        }
+
+        const inactivityDelay = this.config?.sse?.inactivityDelay || 120000
+        this.pageVisibilityHandler = () => {
+            if (!this.config?.sse) {
+                return
+            } else if (document.visibilityState === 'visible') {
+                if (!this.streamingConnection?.isConnected()) {
                     this.logger.debug('Page became visible, refetching config')
                     this.refetchConfig().catch((e) => {
                         this.logger.error('Failed to refetch config', e)
                     })
                     this.streamingConnection?.reopen()
-                } else {
+                }
+                window?.clearTimeout(this.inactivityHandlerId)
+            } else {
+                this.inactivityHandlerId = window?.setTimeout(() => {
```

Comment:
just to be safe you may want to clear the timeout here as well if it exists, before reassigning to a new id. There may be situations where some combination of events causes this to leave a leftover timer

Reply:
I wonder if it's also worth checking specifically for the visibility state being "hidden" rather than using an "else" here.

Reply:
according to this there's only two states, `hidden` and `visible`: https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilityState